project/phpstan.neon.dist

```diff
@@ -0,0 +1,5 @@
+parameters:
```

Comment:
Will this overwrite the existing file?

Reply:
you mean in admin bundle for example? Yes

Reply:
Then we should find a way to create files via dev-kit, if they don't exist

Reply:
If we are using the `DELETE_` prefix, what about `DEFAULT_filename.extension`?

Reply:
What do you mean? Can you make an example?

Reply:
I meant to do something similar to the `DELETE_` prefix, so instead of pushing the `qa.yaml`, pushing `DEFAULT_qa.yaml`. So if the `qa.yaml` exists does nothing, and if it doesn't, it will create the `qa.yaml`.

Reply:
Ok but in this case we would not get any updates.
Let’s say we want to add Tests directory later to the config file... That would be not possible

Reply:
Ok but in this case we would not get any updates.
Let’s say we want to add Tests directory later to the config file... That would be not possible

Reply:
Ok but in this case we would not get any updates.
Let’s say we want to add Tests directory later to the config file... That would be not possible

Reply:
What we can do is exclude the file via projects.yaml for specific bundles

Your case can be done by adding the file, run dev kit and deploy, remove the file from dev kit.

Reply:
@core23 what do you think about it?

Another nice solution would be to extend our `Configuration` for the `projects.yaml` to support all necessary keys of the `phpstan.neon.dist` e.g.

Based on the BlockBundle:
```neon
parameters:
    level: 3

    paths:
        - src
# Uncomment this, when phpstan extensions are allowed
#        - tests

    excludes_analyse:
        - src/Test/BlockServiceTestCase.php
        - tests/bootstrap.php

    autoload_files:
        - vendor/autoload.php

    ignoreErrors:
        - '#Call to an undefined method Sonata\\Form\\Validator\\ErrorElement::assert.*\(\).#'
        - '#Call to an undefined method Symfony\\Component\\Config\\Definition\\Builder\\NodeDefinition::fixXmlConfig\(\).#'
```
->
```yaml
block-bundle:
  phpstan:
      level: 3
      paths:
          - src
      excludes_analyse:
          - src/Test/BlockServiceTestCase.php
          - tests/bootstrap.php
      autoload_files:
          - vendor/autoload.php
      ignoreErrors:
          - '#Call to an undefined method Sonata\\Form\\Validator\\ErrorElement::assert.*\(\).#'
          - '#Call to an undefined method Symfony\\Component\\Config\\Definition\\Builder\\NodeDefinition::fixXmlConfig\(\).#'
  branches:
    master:
      php: ['7.3', '7.4']
      target_php: '7.3'
      versions:
        symfony: ['4.3', '5.0']
    4.x:
      php: ['7.3', '7.4']
      target_php: '7.3'
      versions:
        symfony: ['4.3', '5.0']
```

In this case we can set `level: 1` and `paths: empty array` as default...
A bit more "complex" in the first step but much more flexible.

Reply:
> Another nice solution would be to extend our `Configuration` for the `projects.yaml` to support all necessary keys of the `phpstan.neon.dist` e.g.

I like the `project.yaml` solution, but we should not mix the baseline (ignored errors) and the configuration. The dev-kit shouldn't provide some default `ignore` entries, because not every project is a symfony lib that reports this errors.

Reply:
but the ignored errors are configured per project...

Reply:
> but the ignored errors are configured per project...

IMHO that's intended

Reply:
>  The dev-kit shouldn't provide some default ignore entries, because not every project is a symfony lib that reports this errors.

Sorry I don't understand your sentence here, can you please be more clear what we should do and what not? Thanks

Reply:
We should not do this in the `project.yaml` configuration:

```
      ignoreErrors:
          - '#Call to an undefined method Sonata\\Form\\Validator\\ErrorElement::assert.*\(\).#'
          - '#Call to an undefined method Symfony\\Component\\Config\\Definition\\Builder\\NodeDefinition::fixXmlConfig\(\).#'
```

Providing some default `ignoreErrors` is a bad idea, because then we should also set `reportUnmatchedIgnoredErrors: true`. In this case we will get an outdated ignore list over time. Some projects do not have these errors (e.g. php libs).

Reply:
Ok but why is it in the the block bundle itself? If I configure this:
```yaml
 ignoreErrors:
          - '#Call to an undefined method Sonata\\Form\\Validator\\ErrorElement::assert.*\(\).#'
          - '#Call to an undefined method Symfony\\Component\\Config\\Definition\\Builder\\NodeDefinition::fixXmlConfig\(\).#'
```

for the block bundle, only the block bundle gets this configuration and not any other sonata lib. Or are you saying that it should be nowhere and we should remove it from the config in the block bundle too?

Reply:
Oh sorry, I didn't see that this is for the block bundle only. 

In any case, we should not configure the `ignoreErrors` in any case by the dev-kit, because this will become outdated very soon.

Reply:
> Oh sorry, I didn't see that this is for the block bundle only.

OK

> In any case, we should not configure the ignoreErrors in any case by the dev-kit, because this will become outdated very soon.

But in this case the BlockBundle would fail, right?

Reply:
If the `phpstan.neon.dist` includes a file that does not exist, will it fail? (To include a `baseline.neon`).
(On my phone)

Reply:
Thought about the same

Reply:
It fails if it doesn't exist 😞. But [there is `bootstrap` parameter](https://github.com/phpstan/phpstan#bootstrap-file) which could create the `phpstan-baseline.neon` if it doesn't exist.